PortalSwift/Sources/PortalSwift/Core/Keychain/PortalKeychain.swift

```
@@ -0,0 +1,573 @@
+//
+//  PortalKeychain.swift
+//
+//  Created by Portal Labs, Inc.
+//  Copyright © 2022 Portal Labs, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The main interface for Portal to securely store the client's signing share.
+public class PortalKeychain {
+  public static var metadata: PortalKeychainMetadata?
+
+  public enum KeychainError: Error, Equatable {
+    case clientNotFound
+    case clientIdNotSetYet
+    case itemNotFound(item: String)
+    case itemAlreadyExists(item: String)
+    case keychainUnavailableOrNoPasscode(status: OSStatus)
+    case noAddressForNamespace(_ namespace: PortalNamespace)
+    case noAddressesFound
+    case noWalletForNamespace(_ namespace: PortalNamespace)
+    case noWalletsFound
+    case shareNotFoundForCurve(_ curve: PortalCurve)
+    case unableToEncodeKeychainData
+    case unexpectedItemData(item: String)
+    case unhandledError(status: OSStatus)
+    case unsupportedNamespace(_ chainId: String)
+  }
+
+  public var api: PortalApi? {
+    get { self._api }
+    set(api) {
+      self._api = api
+
+      //  Load the metadata as soon as the api is set
+      if let _ = _api {
+        Task {
+          do {
+            try await self.loadMetadata()
+          } catch {}
```

How come we fail silently here?